lib/puppet/module.rb

```diff
@@ -428,22 +428,24 @@ def initialize_i18n
 
     locales_path = File.absolute_path('locales', path)
 
-    begin
-      Puppet::GettextConfig.initialize(locales_path, :po)
+    if Puppet::GettextConfig.initialize(locales_path, :po)
       Puppet.debug "#{module_name} initialized for i18n: #{GettextSetup.translation_repositories[module_name]}"
-    rescue
-      config_path = File.absolute_path('config.yaml', locales_path)
-      Puppet.debug "Could not find locales configuration file for #{module_name} at #{config_path}. Skipping i18n initialization."
+    else if Puppet::GettextConfig.gettext_loaded?
+        config_path = File.absolute_path('config.yaml', locales_path)
+        Puppet.debug "Could not find locales configuration file for #{module_name} at #{config_path}. Skipping i18n initialization."
```

Comment:
I'm going to come back to this one as I don't think it's quite the same as before.

Previously we did a `rescue` to capture the `NoConfigFoundError` raised at https://github.com/puppetlabs/gettext-setup-gem/blob/master/lib/gettext-setup/gettext_setup.rb#L27 when `config.yaml` doesn't exist.

Now we no longer catch the error, and then create a message about a missing `config.yaml` - which doesn't align with what is actually happening.  If `config.yaml` is missing, this `else` is never hit given the exception is raised, and we bubble up to the next matching exception handler in the call stack.

Reply:
Yeah, checked that this doesn't fail gracefully if `locales` exists but doesn't contain `config.yaml`.

```
[root@10-32-167-127 ~]# puppet agent -t
Info: Using configured environment 'production'
Info: Retrieving pluginfacts
Error: /File[/opt/puppetlabs/puppet/cache/facts.d]: Failed to generate additional resources using 'eval_generate': Error 500 on SERVER: Server Error: No config.yaml found! (searching: /etc/puppetlabs/code/environments/production/site/dummy1/locales/config.yaml)
Error: /File[/opt/puppetlabs/puppet/cache/facts.d]: Could not evaluate: Could not retrieve file metadata for puppet:///pluginfacts: Error 500 on SERVER: Server Error: No config.yaml found! (searching: /etc/puppetlabs/code/environments/production/site/dummy1/locales/config.yaml)
Info: Retrieving plugin
Error: /File[/opt/puppetlabs/puppet/cache/lib]: Failed to generate additional resources using 'eval_generate': Error 500 on SERVER: Server Error: No config.yaml found! (searching: /etc/puppetlabs/code/environments/production/site/dummy1/locales/config.yaml)
Error: /File[/opt/puppetlabs/puppet/cache/lib]: Could not evaluate: Could not retrieve file metadata for puppet:///plugins: Error 500 on SERVER: Server Error: No config.yaml found! (searching: /etc/puppetlabs/code/environments/production/site/dummy1/locales/config.yaml)
Info: Loading facts
Error: Could not retrieve catalog from remote server: Error 500 on SERVER: Server Error: No config.yaml found! (searching: /etc/puppetlabs/code/environments/production/site/dummy1/locales/config.yaml) on node 10-32-167-127.rfc1918.puppetlabs.net
Warning: Not using cache on failed catalog
Error: Could not retrieve catalog; skipping run
```

@Magisus is looking at fixing it.

Reply:
I'm OK with the server 500 error / message as seen above.

If it wasn't clear, my point is just that the `else` message is incorrect / misleading given the conditions necessary to get here.

Reply:
PR to update what we check for existence: https://github.com/puppetlabs/puppet/pull/6259 This method will now return false if the file doesn't exist, rather than throw.